projects/angular-gridster2/src/lib/gridsterResizable.service.ts

```diff
@@ -807,4 +823,90 @@ export class GridsterResizable {
       width + 'px'
     );
   }
+  /**
+   * Enforces the aspect ratio by recalculating grid positions based on current pixel dimensions
+   */
+  private enforceAspectRatio(): void {
+    const aspectRatio = this.gridsterItem.$item.itemAspectRatio || this.gridster.$options.itemAspectRatio;
+    if (!aspectRatio) {
+      return;
+    }
+
+    // Calculate the target aspect ratio (cols/rows)
+    const targetRatio = aspectRatio;
```

Comment:
I think if we multiply by ratio of colWidth / rowHeight it should work, and we can apply the ratio and correct during the resize the width/height of the item
`const targetRatio = aspectRatio * (this.gridster.curColWidth / this.gridster.curRowHeight);`